libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -311,7 +311,7 @@ void AP_ESC_Telem::send_esc_telemetry_mavlink(uint8_t mav_chan)
     const uint8_t esc_offset = constrain_int16(mavlink_offset, 0, ESC_TELEM_MAX_ESCS-1);
     const uint8_t num_idx = ESC_TELEM_MAX_ESCS/4;
     for (uint8_t idx = 0; idx < num_idx; idx++) {
-        const uint8_t i = (next_idx + idx) % num_idx;
+        const uint8_t i = num_idx > 0 ? (next_idx + idx) % num_idx : 0;
```

Comment:
How can this ever happen @magicrub? Surely you get no iterations of the loop if num_idx is 0?

Reply:
oh, now I understand your question. You're right, the current code is actually wrong. If we only have 1, 2 or 3 PWMs then we don't even send   MSG_ESC_TELEMETRY_1_TO_4. The issue I had was a compiler warning (hence error) for a mod of 0. Seems like this whole thing needs to get reworked a little in general